src/piggui.rs

```diff
@@ -191,12 +191,16 @@ impl Application for Piggui {
             ConnectRequest(new_target) => {
                 // Show spinner when connection requested
                 self.connect_dialog.show_spinner = true;
+                // Disable widgets when connection requested
+                self.connect_dialog.disable_widgets = true;
```

Comment:
Couldn't this also be done internally in dialog, when the button is pressed and message sent, and avoid making public and piggui.rs having to know about it?

(Same as another comment ...)